src/net_processing.cpp

```diff
@@ -1273,7 +1273,7 @@ void PeerManagerImpl::FinalizeNode(const CNode& node)
     }
     } // cs_main
     if (node.fSuccessfullyConnected && misbehavior == 0 &&
-        !node.IsBlockOnlyConn() && !node.IsInboundConn()) {
+        !node.IsBlockOnlyConn() && !node.IsManualBlockOnlyConn() && !node.IsInboundConn()) {
```

Comment:
`IsBlockOnlyConn` and `IsManualConn` (below) should just return true for `MANUAL_BLOCK_RELAY` rather than additional checks every place

Reply:
I didn't do that because `IsBlockOnlyConn()` is used in several places (e.g. `PeerManagerImpl::EvictExtraOutboundPeers`, `CConnman::GetExtraBlockRelayCount()`, `GetCurrentBlockRelayOnlyConns()`) which would not apply to manual block-relay-only connections.